Source/Core/Core/IOS/DI/DI.h

```diff
@@ -25,16 +37,111 @@ class DI : public Device
 public:
   DI(Kernel& ios, const std::string& device_name);
 
+  static void InterruptFromDVDInterface(DVDInterface::DIInterruptType interrupt_type);
+  static DiscIO::Partition GetCurrentPartition();
+
   void DoState(PointerWrap& p) override;
 
+  IPCCommandResult Open(const OpenRequest& request) override;
   IPCCommandResult IOCtl(const IOCtlRequest& request) override;
   IPCCommandResult IOCtlV(const IOCtlVRequest& request) override;
 
-  void FinishIOCtl(DVDInterface::DIInterruptType interrupt_type);
+  enum class DIIoctl : u32
+  {
+    DVDLowInquiry = 0x12,
+    DVDLowReadDiskID = 0x70,
+    DVDLowRead = 0x71,
+    DVDLowWaitForCoverClose = 0x79,
+    DVDLowGetCoverRegister = 0x7a,  // DVDLowPrepareCoverRegister
+    DVDLowNotifyReset = 0x7e,
+    DVDLowSetSpinupFlag = 0x7f,
+    DVDLowReadDvdPhysical = 0x80,
+    DVDLowReadDvdCopyright = 0x81,
+    DVDLowReadDvdDiscKey = 0x82,
+    DVDLowGetLength = 0x83,
+    DVDLowGetImmBuf = 0x84,  // Unconfirmed name
+    DVDLowUnmaskCoverInterrupt = 0x85,
+    DVDLowClearCoverInterrupt = 0x86,
+    // 0x87 is a dummied out command
+    DVDLowGetCoverStatus = 0x88,
+    DVDLowEnableCoverInterrupt = 0x89,  // Unconfirmed name
+    DVDLowReset = 0x8a,
+    DVDLowOpenPartition = 0x8b,  // ioctlv only
+    DVDLowClosePartition = 0x8c,
+    DVDLowUnencryptedRead = 0x8d,
+    DVDLowEnableDvdVideo = 0x8e,
+    DVDLowGetNoDiscOpenPartitionParams = 0x90,       // ioctlv, dummied out
+    DVDLowNoDiscOpenPartition = 0x91,                // ioctlv, dummied out
+    DVDLowGetNoDiscBufferSizes = 0x92,               // ioctlv, dummied out
+    DVDLowOpenPartitionWithTmdAndTicket = 0x93,      // ioctlv
+    DVDLowOpenPartitionWithTmdAndTicketView = 0x94,  // ioctlv
+    DVDLowGetStatusRegister = 0x95,                  // DVDLowPrepareStatusRegsiter
+    DVDLowGetControlRegister = 0x96,                 // DVDLowPrepareControlRegister
+    DVDLowReportKey = 0xa4,
+    // 0xa8 is unusable externally
+    DVDLowSeek = 0xab,
+    DVDLowReadDvd = 0xd0,
+    DVDLowReadDvdConfig = 0xd1,
+    DVDLowStopLaser = 0xd2,
+    DVDLowOffset = 0xd9,
+    DVDLowReadDiskBca = 0xda,
+    DVDLowRequestDiscStatus = 0xdb,
+    DVDLowRequestRetryNumber = 0xdc,
+    DVDLowSetMaximumRotation = 0xdd,
+    DVDLowSerMeasControl = 0xdf,
+    DVDLowRequestError = 0xe0,
+    DVDLowAudioStream = 0xe1,
+    DVDLowRequestAudioStatus = 0xe2,
+    DVDLowStopMotor = 0xe3,
+    DVDLowAudioBufferConfig = 0xe4
+  };
+
+  enum class DIResult : s32
+  {
+    SUCCESS = 1,
```

Comment:
Would that also apply to `GPIO`, `DIInterruptType`, and `DICommand`?

Reply:
You can use uppercase for all the letters of an acronym. So there's no problem with the name `DIResult` or the other three names you mentioned, but `SUCCESS` should be `Success`.

Reply:
I'm asking about the members of `DICommand` and the like.  This is what `DICommand` currently looks like:

https://github.com/dolphin-emu/dolphin/blob/0ac8ceda411806e17886ec0fa58b5344775cfe8f/Source/Core/Core/HW/DVD/DVDInterface.h#L27-L52

(I'm guessing, though, that it's fine to leave GPIO and DIInterruptType members in all caps, since at least the common terminology for them is in all caps?)

Reply:
Oh, the members? For `DICommand`, they should be like `Inquiry`, `ReportKey`, `ReadDVDMetadata`, etc. I would be fine with leaving the `DIInterruptType` members in all caps. I know less about the members in `GPIO`, but if that's how they usually are written, I guess that's fine? (Especially considering the ones that don't contain full words, like `EEP_CLK`.)